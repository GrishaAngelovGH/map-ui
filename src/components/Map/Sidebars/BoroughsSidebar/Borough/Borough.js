import { useEffect, useState } from 'react'

const Borough = ({ id, active, name, code, areaHectares, coordinates, onClick }) => {
  const [isFavorite, setIsFavorite] = useState(false)

  const handleViewOnMapClick = () => {
    const coords = coordinates.map(([lng, lat]) => [lat, lng])

    onClick(id, coords)
  }

  const handleAddToFavoritesClick = () => {
    const favoriteBoroughs = window.localStorage.getItem('boroughs')

    if (!favoriteBoroughs.includes(name)) {
      window.localStorage.setItem('boroughs', `${favoriteBoroughs}${name},`)
    }

    setIsFavorite(true)
  }

  const handleRemoveFromFavoritesClick = () => {
    const favoriteBoroughs = window.localStorage.getItem('boroughs').split(',')
    const newFavoriteBoroughs = favoriteBoroughs.filter(v => v !== name)
    window.localStorage.setItem('boroughs', newFavoriteBoroughs)
    setIsFavorite(false)
  }

  useEffect(() => {
    const favoriteBoroughs = window.localStorage.getItem('boroughs')

    if (favoriteBoroughs.includes(name)) {
      setIsFavorite(true)
    }
  }, [])

  return (
    <div className={`row mb-3 ${active ? 'border border-1 border-primary' : 'border'} rounded bg-light p-2`}>
      <div className='col-md-12'>
        <h5 className='fw-bold'>#{id.toString().padStart(2, '0')} {name}</h5>

        <h5>Code: {code}</h5>

        <h5>Area Hectares: {areaHectares}</h5>

        <button className='btn btn-primary w-100' onClick={handleViewOnMapClick}>View on Map</button>

        <button
          className={`btn  ${isFavorite ? 'btn-warning' : 'btn-outline-warning'} w-100 mt-2`}
          onClick={isFavorite ? handleRemoveFromFavoritesClick : handleAddToFavoritesClick}
        >
          {isFavorite ? 'Remove from Favorites' : 'Save as Favorite'}
        </button>
      </div>
    </div >
  )
}

export default Borough