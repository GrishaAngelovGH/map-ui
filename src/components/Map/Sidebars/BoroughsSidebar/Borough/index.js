import Borough from './Borough'

export default Borough