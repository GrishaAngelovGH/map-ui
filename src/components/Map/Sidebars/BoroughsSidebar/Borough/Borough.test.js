import { render } from '@testing-library/react'

import Borough from './Borough'

const localStorageMock = (function () {
  let store = {};

  return {
    getItem(key) {
      return store[key];
    },

    setItem(key, value) {
      store[key] = value;
    }
  }
})()

Object.defineProperty(window, 'localStorage', { value: localStorageMock })

test('should render Borough component marked as favorite', () => {
  window.localStorage.setItem('boroughs', 'borough')

  const view = render(
    <Borough
      id={1}
      name='borough'
      code='123a'
      areaHectares={10}
    />
  )

  expect(view).toMatchSnapshot()
})

test('should render Borough component not marked as favorite', () => {
  window.localStorage.setItem('boroughs', '')

  const view = render(
    <Borough
      id={1}
      name='borough'
      code='123a'
      areaHectares={10}
    />
  )

  expect(view).toMatchSnapshot()
})