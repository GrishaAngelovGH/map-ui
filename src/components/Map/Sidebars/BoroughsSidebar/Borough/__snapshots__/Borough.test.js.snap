// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Borough component marked as favorite 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row mb-3 border rounded bg-light p-2"
      >
        <div
          class="col-md-12"
        >
          <h5
            class="fw-bold"
          >
            #
            01
             
            borough
          </h5>
          <h5>
            Code: 
            123a
          </h5>
          <h5>
            Area Hectares: 
            10
          </h5>
          <button
            class="btn btn-primary w-100"
          >
            View on Map
          </button>
          <button
            class="btn  btn-warning w-100 mt-2"
          >
            Remove from Favorites
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row mb-3 border rounded bg-light p-2"
    >
      <div
        class="col-md-12"
      >
        <h5
          class="fw-bold"
        >
          #
          01
           
          borough
        </h5>
        <h5>
          Code: 
          123a
        </h5>
        <h5>
          Area Hectares: 
          10
        </h5>
        <button
          class="btn btn-primary w-100"
        >
          View on Map
        </button>
        <button
          class="btn  btn-warning w-100 mt-2"
        >
          Remove from Favorites
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render Borough component not marked as favorite 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row mb-3 border rounded bg-light p-2"
      >
        <div
          class="col-md-12"
        >
          <h5
            class="fw-bold"
          >
            #
            01
             
            borough
          </h5>
          <h5>
            Code: 
            123a
          </h5>
          <h5>
            Area Hectares: 
            10
          </h5>
          <button
            class="btn btn-primary w-100"
          >
            View on Map
          </button>
          <button
            class="btn  btn-outline-warning w-100 mt-2"
          >
            Save as Favorite
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row mb-3 border rounded bg-light p-2"
    >
      <div
        class="col-md-12"
      >
        <h5
          class="fw-bold"
        >
          #
          01
           
          borough
        </h5>
        <h5>
          Code: 
          123a
        </h5>
        <h5>
          Area Hectares: 
          10
        </h5>
        <button
          class="btn btn-primary w-100"
        >
          View on Map
        </button>
        <button
          class="btn  btn-outline-warning w-100 mt-2"
        >
          Save as Favorite
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
